tests/test_decision_making/test_acquisition_functions/test_thompson_sampling.py

```diff
@@ -0,0 +1,218 @@
+    datasets = {OBJECTIVE: dataset}
+    key = jr.PRNGKey(42)
+    ts_acquisition_builder = ThompsonSampling(num_rff_features=500)
+    ts_acquisition_function = ts_acquisition_builder.build_acquisition_function(
+        posteriors=posteriors, datasets=datasets, key=key
+    )
+    if dimensionality == 1:
+        test_X = generate_dummy_1d_test_X(num_test_points)
+    elif dimensionality == 2:
+        test_X = generate_dummy_2d_test_X(int(jnp.sqrt(num_test_points)))
+    ts_acquisition_function_values = ts_acquisition_function(test_X)
+    assert ts_acquisition_function_values.shape == (num_test_points, 1)
+
+
+def test_thompson_sampling_acquisition_function_same_key_same_function():
+    dataset = generate_dummy_1d_dataset()
+    posterior = generate_dummy_conjugate_posterior(dataset)
+    posteriors = {OBJECTIVE: posterior}
+    datasets = {OBJECTIVE: dataset}
+    key = jr.PRNGKey(42)
+    ts_acquisition_builder_one = ThompsonSampling(num_rff_features=500)
+    ts_acquisition_builder_two = ThompsonSampling(num_rff_features=500)
+    ts_acquisition_function_one = ts_acquisition_builder_one.build_acquisition_function(
+        posteriors=posteriors, datasets=datasets, key=key
+    )
+    ts_acquisition_function_two = ts_acquisition_builder_two.build_acquisition_function(
+        posteriors=posteriors, datasets=datasets, key=key
+    )
+    test_X = jnp.linspace(0.0, 10.0, 100).reshape(-1, 1)
+    ts_acquisition_function_one_values = ts_acquisition_function_one(test_X)
+    ts_acquisition_function_two_values = ts_acquisition_function_two(test_X)
+    assert isinstance(ts_acquisition_function_one, Callable)
+    assert isinstance(ts_acquisition_function_two, Callable)
+    assert (
+        ts_acquisition_function_one_values == ts_acquisition_function_two_values
+    ).all()
+
+
+def test_thompson_sampling_acquisition_function_different_key_different_function():
+    dataset = generate_dummy_1d_dataset()
+    posterior = generate_dummy_conjugate_posterior(dataset)
+    posteriors = {OBJECTIVE: posterior}
+    datasets = {OBJECTIVE: dataset}
+    key_one = jr.PRNGKey(42)
+    key_two = jr.PRNGKey(43)
+    ts_acquisition_builder = ThompsonSampling(num_rff_features=500)
+    ts_acquisition_function_one = ts_acquisition_builder.build_acquisition_function(
+        posteriors=posteriors, datasets=datasets, key=key_one
+    )
+    ts_acquisition_function_two = ts_acquisition_builder.build_acquisition_function(
+        posteriors=posteriors, datasets=datasets, key=key_two
+    )
+    test_X = jnp.linspace(0.0, 10.0, 100).reshape(-1, 1)
+    ts_acquisition_function_one_values = ts_acquisition_function_one(test_X)
+    ts_acquisition_function_two_values = ts_acquisition_function_two(test_X)
+    assert isinstance(ts_acquisition_function_one, Callable)
+    assert isinstance(ts_acquisition_function_two, Callable)
+    assert not (
+        ts_acquisition_function_one_values == ts_acquisition_function_two_values
+    ).all()
```

Comment:
You could do a monte-carlo test here?

i.e. eval a  bunch of TS at a set of points X then check that the avg over the TS matches with the posterior mean over X?

Reply:
Decided against doing this in the end as the test would take too long to run, and there is already a monte-carlo test for testing out the quality of approximate samples.